src/app/pages/account/groups/group-form/group-form.component.ts

```diff
@@ -158,6 +169,81 @@ export class GroupFormComponent implements OnInit {
     });
   }
 
+  private togglePrivilegesForGroup(groupId: number): Observable<Privilege[]> {
+    const requests$: Observable<Privilege>[] = [];
+
+    if (this.selectedPrivilegeIds) {
+      const privileges = this.privilegesList
+        .filter((privilege) => this.selectedPrivilegeIds.some((privilegeId) => privilege.id === privilegeId));
+
+      privileges.forEach((privilege) => {
+        requests$.push(
+          this.ws.call('privilege.update', [
+            privilege.id,
+            this.mapPrivilegeToPrivilegeUpdate(
+              privilege,
+              [...privilege.local_groups.map((group) => group.gid), groupId],
+            ),
+          ]),
+        );
+      });
+    }
+
+    if (this.existingPrivilegesRemoved) {
+      const privileges = this.privilegesList
+        .filter((privilege) => this.existingPrivilegesRemoved.some((privilegeId) => privilege.id === privilegeId));
```

Comment:
Same here

Reply:
@RehanY147  - how do you expect to know which privilege was removed?
We need to also update the PRIVILEGE where the group was removed from the privilege.